### PR TITLE
Refactoring/exit startup power

### LINKED ISSUE #134: Improve stall protection

### PR DESCRIPTION
**Pull request by damosvil**
Refactored exit_startup_power to improve coherency.
Updated switch_power_off to also set Pwm_Limits to zero to enforce shut down
Renamed Pwm_Limit to Pwm_Limit_Startup_n_Temp to make more clear the use of this variable

### CODE REVIEW

File: src/Bluejay.asm [hunk 1]
```diff
@@ -1082,16 +1079,13 @@ exit_run_mode_is_stall:
 
 exit_run_mode_is_stall_beep:
     ; Stalled too many times
-    clr  IE_EA
+    ;clr  IE_EA                         ; Interrupts already disabled above
```
Comment (stylesuxx) [inline src/Bluejay.asm:1082]: I'd remove this line completely, maybe just move the comment up that interrupts are already disabled at this points.

File: src/Modules/Isrs.asm [hunk 2]
```diff
@@ -347,7 +347,7 @@ t1_int_zero_rcp_checked:
 t1_int_zero_rcp_checked_set_limit:
     ; Set pwm limit
     clr  C
-    mov  A, Pwm_Limit                   ; Limit to the smallest
+    mov  A, Pwm_Limit_Startup_n_Temp                   ; Limit to the smallest
```
Comment (stylesuxx) [inline src/Modules/Isrs.asm:350]: Comment indentation.

File: src/Modules/Power.asm [hunk 3]
```diff
@@ -35,9 +35,16 @@
 ;
 ;**** **** **** **** **** **** **** **** **** **** **** **** ****
 switch_power_off:
+    ; This three macros disable MOSFET communtation
+    ; By them any dshot frame won't have effect until commutation starts
```
Comment (stylesuxx) [inline src/Modules/Power.asm:39]: This comment is a bit unclear, maybe we could re-phrase:
> Dshot frames will be ignored after commutation is started again at motor_start_bidir_done

**Review by stylesuxx — CHANGES_REQUESTED** [review]

File: src/Bluejay.asm (hunk 1 above)
Comment (damosvil) [reply src/Bluejay.asm]: Removed

File: src/Modules/Isrs.asm (hunk 2 above)
Comment (damosvil) [reply src/Modules/Isrs.asm]: Updated indentation

File: src/Modules/Power.asm (hunk 3 above)
Comment (damosvil) [reply src/Modules/Power.asm]: Updated wording

File: src/Bluejay.asm [hunk 4]
```diff
@@ -929,7 +929,7 @@ initial_run_phase_done:
     ; This plus the power limits applied in set_pwm_limit function
     ; act as a startup power limiter to protect the esc and the motor
     ; during startup, jams produced after crashes and desyncs recovery
-    mov  Pwm_Limit, #255                ; Reset temperature level pwm limit
+    mov  Pwm_Limit_Startup_n_Temp, #255                ; Reset temperature level pwm limit
```
Comment (stylesuxx) [inline src/Bluejay.asm:932]: Comment indentation.

File: src/Bluejay.asm [hunk 5]
```diff
@@ -1068,9 +1068,6 @@ ENDIF
     ljmp motor_start                    ; Go back and try starting motors again
 
 exit_run_mode_is_stall:
-    ; Enable all interrupts (disabled above, in exit_run_mode)
-    setb IE_EA
```
Comment (hyp0dermik-code) [inline src/Bluejay.asm:1072]: Can the tidyup of interrupt handling be included in the comment message too please?

Comment (damosvil) [reply src/Bluejay.asm:1072]: In principle that code has been removed. So you mean to the commit message?

File: src/Bluejay.asm (hunk 4 above)
Comment (damosvil) [reply src/Bluejay.asm]: Fixed

File: src/Bluejay.asm (hunk 5 above)
Comment (hyp0dermik-code) [reply src/Bluejay.asm:1072]: Yes my apologies

Comment (damosvil) [reply src/Bluejay.asm:1072]: Done
imagen

Comment (hyp0dermik-code) [reply src/Bluejay.asm:1072]: Squashing commits not be a bad idea for the future, I think @stylesuxx can turn this on on the repo side

**Review by hyp0dermik-code — APPROVED** [review]

**Review by stylesuxx — APPROVED** [review]

**Pull request merged**